p2p/server.go

```diff
@@ -405,6 +407,19 @@ func (srv *Server) PeerCount() int {
 	return count
 }
 
+// InboundCount returns the number of inbound peers.
+func (srv *Server) InboundCount() int {
```

Comment:
See my comment regarding adding the incoming peer as a validator peer in the handshake. If that's done, then function doesn't need to be created.